### PR TITLE
Implement .env util using Azure KeyVaults

### PR DESCRIPTION
**Pull request by t-kobayashi**
ref.  https://gitlab.com/vc-robotics/vrj-cloud/infrastructure/issues/3 (sensyn-robotics internal)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+func (f *fetcher) fetch(rawurl string) (string, error) {
+	url, err := url.Parse(rawurl)
+	if err != nil {
+		return "", err
+	}
+	if !strings.HasSuffix(url.Hostname(), "vault.azure.net") {
+		return "", fmt.Errorf("Invalid url - %s", rawurl)
+	}
+	b, err := f.getToken()
+	req, err := http.NewRequest("GET", rawurl+"?api-version=7.0", nil)
```
Comment (akayaman) [inline main.go:64]: in the above two lines, you don't need to check the `err` variable ?

File: main.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	}
+
+	return result.Value, nil
+}
+
+func (f *fetcher) getToken() (string, error) {
+	if f.token != "" {
+		return f.token, nil
+	}
+	req, err := http.NewRequest("GET", "http://169.254.169.254/metadata/identity/oauth2/token?api-version=2019-06-04&resource=https%3A%2F%2Fvault.azure.net", nil)
```
Comment (akayaman) [inline main.go:93]: Could tell me what `http://169.254.169.254` means ?

Comment (t-kobayashi) [reply main.go:93]: `169.254.169.254` is link-local address ,  used for getting  instance metadata in Azure, (AWS too).

https://docs.microsoft.com/azure/virtual-machines/windows/instance-metadata-service#usage

File: main.go (hunk 1 above)
Comment (t-kobayashi) [reply main.go:64]: Thanks, I will fix this.

Comment (t-kobayashi) [reply main.go:64]: fixed

File: main.go (hunk 2 above)
Comment (akayaman) [reply main.go:93]: I understand. thank you.

**Pull request merged**